File: add_audit_admin_table.py
import sqlite3
import os
from datetime import datetime

# Open the database
db_path = os.path.join(os.getcwd(), 'mazaya.db')
conn = sqlite3.connect(db_path)
cursor = conn.cursor()

print(f"Connected to database: {db_path}")

# Create audit_admin_users table
try:
    cursor.execute('''
        CREATE TABLE IF NOT EXISTS audit_admin_users (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            email TEXT NOT NULL UNIQUE,
            created_at TEXT,
            updated_at TEXT
        )
    ''')
    print("audit_admin_users table created or already exists.")
    
    # Add the user as the first audit admin
    admin_email = 'yousef.baitalmal@flyadeal.com'
    now = datetime.now().isoformat()
    
    try:
        cursor.execute('''
            INSERT OR IGNORE INTO audit_admin_users (email, created_at, updated_at)
            VALUES (?, ?, ?)
        ''', (admin_email, now, now))
        
        if cursor.rowcount > 0:
            print(f"Added {admin_email} as audit admin.")
        else:
            print(f"{admin_email} is already an audit admin.")
            
        conn.commit()
    except sqlite3.Error as e:
        print(f"Error adding initial audit admin: {e}")
        
except sqlite3.Error as e:
    print(f"Error creating audit_admin_users table: {e}")
finally:
    conn.close()
    print("Database connection closed.")

File: add_audit_admin_table.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

// Open the database
const dbPath = path.resolve(process.cwd(), 'mazaya.db');
const db = new sqlite3.Database(dbPath, (err) => {
  if (err) {
    console.error('Error opening database:', err.message);
    process.exit(1);
  }
  console.log('Connected to the database.');
});

// Create audit_admin_users table
db.run(`
  CREATE TABLE IF NOT EXISTS audit_admin_users (
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    email TEXT NOT NULL UNIQUE,
    created_at TEXT,
    updated_at TEXT
  )
`, (err) => {
  if (err) {
    console.error('Error creating audit_admin_users table:', err.message);
    db.close();
    process.exit(1);
  }
  console.log('audit_admin_users table created or already exists.');
  
  // Add the user as the first audit admin
  const adminEmail = 'yousef.baitalmal@flyadeal.com';
  const now = new Date().toISOString();
  
  db.run(`
    INSERT OR IGNORE INTO audit_admin_users (email, created_at, updated_at)
    VALUES (?, ?, ?)
  `, [adminEmail, now, now], (err) => {
    if (err) {
      console.error('Error adding initial audit admin:', err.message);
    } else {
      console.log(`Added ${adminEmail} as audit admin.`);
    }
    
    // Close the database
    db.close((err) => {
      if (err) {
        console.error('Error closing database:', err.message);
      } else {
        console.log('Database connection closed.');
      }
    });
  });
});